centreon/www/include/Administration/myAccount/DB-Func.php

```diff
@@ -258,3 +258,77 @@ function checkAutologinValue(array $fields)
 
     return count($errors) > 0 ? $errors : true;
 }
+function updateContactInDBIfLocal($contact_id = null){
+    global $pearDB, $centreon, $form;
+
+    if (!$contact_id){
+        return ;
+    }
+    $query = "SELECT contact_name, contact_alias, contact_email, contact_auth_type FROM contact WHERE contact_id = :contact_id";
+    $stmt = $pearDB->prepare($query);
+    $stmt->bindValue(':contact_id', $contact_id, PDO::PARAM_INT);
+    $stmt->execute();
+    $DBRESULT = $stmt->fetch(PDO::FETCH_ASSOC);
+    if($DBRESULT['contact_auth_type'] !== 'local') {
+        $ret = $form->getSubmitValues();
+        $ret['contact_pager'] = !empty($ret['contact_pager']) ?
+            CentreonUtils::escapeSecure($ret['contact_pager'], CentreonUtils::ESCAPE_ILLEGAL_CHARS) : '';
+        $ret['contact_autologin_key'] = !empty($ret['contact_autologin_key']) ?
+            CentreonUtils::escapeSecure($ret['contact_autologin_key'], CentreonUtils::ESCAPE_ILLEGAL_CHARS) : '';
+        $ret['contact_lang'] = !empty($ret['contact_lang']) ?
+            CentreonUtils::escapeSecure($ret['contact_lang'], CentreonUtils::ESCAPE_ILLEGAL_CHARS) : '';
+
+        $rq = 'UPDATE contact SET ' .
+            'contact_location = :contactLocation, ' .
+            'contact_lang = :contactLang, ' .
+            'contact_pager = :contactPager, ' .
+            'default_page = :defaultPage, ' .
+            'show_deprecated_pages = :showDeprecatedPages, ' .
+            'contact_autologin_key = :contactAutologinKey, ' .
+            'contact_theme = :contactTheme';
+        $rq .= ' WHERE contact_id = :contactId';
+
+        $stmt = $pearDB->prepare($rq);
+        $stmt->bindValue(':contactLang', $ret['contact_lang'], \PDO::PARAM_STR);
+        $stmt->bindValue(
+            ':contactPager',
+            !empty($ret['contact_pager']) ? $ret['contact_pager'] : null,
+            \PDO::PARAM_STR
+        );
+        $stmt->bindValue(
+            ':contactAutologinKey',
+            !empty($ret['contact_autologin_key']) ? $ret['contact_autologin_key'] : null,
+            \PDO::PARAM_STR
+        );
+        $stmt->bindValue(
+            ':contactLocation',
+            !empty($ret['contact_location']) ? $ret['contact_location'] : null,
+            \PDO::PARAM_INT
+        );
+        $stmt->bindValue(
+            ':contactTheme',
+            !empty($ret['contact_theme']['contact_theme']) ? $ret['contact_theme']['contact_theme'] : "light",
+            \PDO::PARAM_STR
+        );
+        $stmt->bindValue(':defaultPage', !empty($ret['default_page']) ? $ret['default_page'] : null, \PDO::PARAM_INT);
+        $stmt->bindValue(':showDeprecatedPages', isset($ret['show_deprecated_pages']) ? 1 : 0, \PDO::PARAM_STR);
+        $stmt->bindValue(':contactId', $contact_id, \PDO::PARAM_INT);
+        $stmt->execute();
+        $stmt->closeCursor();
+        if (isset($ret["contact_passwd"]) && !empty($ret["contact_passwd"])) {
```

Comment:
`empty` is already doing an `isset` check internally.
```suggestion
        if (!empty($ret["contact_passwd"])) {
```